src/charm.py

```diff
@@ -43,7 +44,7 @@ def __init__(self, *args: typing.Any) -> None:
             flask_container=self.unit.get_container(FLASK_CONTAINER_NAME),
         )
         self._flask_app = FlaskApp(charm_state=self._charm_state)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.config_changed, self.on_config_changed)
```

Comment:
What is the reason for changing this? The event handlers should be private methods with a certain naming conventions, ISD014 specifies that in more detail

Reply:
I'm calling them from the database module (and I have to do that), so they are not private anymore

Reply:
Ah I didn't notice that, I have left another comment to discuss that